### DISCUSSION TITLE
Question about dsc.keypadFireAlarm, keypadAuxAlarm and keypadPanicAlarm

### DISCUSSION THREAD

**Discussion by jimtng** [taligentx/dscKeybusInterface#83] [dscKeybusInterface Discussion]
Hi,

I noticed that these three flags (keypadFireAlarm, keypadAuxAlarm and keypadPanicAlarm) don't have the corresponding "Changed" flag. When do we know that these alarms have been cleared?

**Comment by taligentx** [comment]
These are one-shot commands, the panel does not repeat the command so setting each of these flags to `false` is enough to track changes.  All other commands are constantly repeated by the panel so there a second flag is needed to track changes.

**Comment by jimtng** [comment]
I still don't understand it. What is the purpose/function of those three "keypad xxx alarms" ? Whey do they get set / sent by the panel, and what does it mean?

**Comment by taligentx** [comment]
These are for the physical keypad panic keys so you can determine if a user has physically pressed one of the three panic buttons on one of the keypads.

**Comment by otto76** [comment]
Is it possible to trigger Panic alarm?
And can we trigger 'full alarm' including sirena, etc. Panic button sends silent call to security service w/o screaming

**Comment by jwolansk** [comment]
I second this, triggering the alarms would be very nice to have.

**Comment by taligentx** [comment]
> Is it possible to trigger Panic alarm?
> And can we trigger 'full alarm' including sirena, etc. Panic button sends silent call to security service w/o screaming

This is already possible, check the virtual keypad section in `README.md`.  By default the panic key is silent but you can configure the DSC panel to turn on the bell (siren) when the panic key is pressed by turning on section `015` option 2 - Panic Key Audible (Bell / Beeps).

**Reply by hernanigarza** [reply]
Hello, is it possible to send the panic command by MQTT?

**Reply by taligentx** [reply]
Hi @hernanigarza - I've updated the HomeAssistant examples in the `develop` branch to show how to add a button card in HomeAssistant that sends "P" to the MQTT subscribe topic, and updated the sketch to look for this payload and trigger the panic button.  The same method can be used for the other keypad alarms.
